flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/TieredStorageProducerClient.java

```diff
@@ -42,6 +45,17 @@ public class TieredStorageProducerClient {
 
     private final List<TierProducerAgent> tierProducerAgents;
 
+    /** The current writing segment index for each subpartition. */
+    private final int[] currentSubpartitionSegmentId;
+
+    /** The current writing tier index for each subpartition. */
+    private final int[] currentSubpartitionTierIndex;
+
+    /** Indicate whether the segment for each subpartition is finished. */
+    private final boolean[] isSubpartitionSegmentFinished;
```

Comment:
`isCurrentSubpartitionSegmentFinished`

Reply:
This field is removed because the logic of EndOfSegment has been changed.